python/lsst/daf/butler/core/connectionString.py

```diff
@@ -0,0 +1,168 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("ConnectionString", "ConnectionStringFactory")
+
+from sqlalchemy.exc import NoSuchModuleError
+
+from sqlalchemy.engine import url
+from lsst.daf.butler.core.dbAuth import DbAuth
+from lsst.daf.butler.core.registry import RegistryConfig
+
+DB_AUTH_ENVVAR = "LSST_DB_AUTH"
+"""Default name of the environmental variable that will be used to locate DB
+credentials configuration file. """
+
+DB_AUTH_PATH = "~/.lsst/db-auth.yaml"
+"""Default path at which it is expected that DB credentials are found."""
+
+
+class ConnectionString(url.URL):
+    """Builds a connection string of the format:
+
+        dialect+driver://user:password@host:port/database
+
+    by parsing (in order of precedence):
+
+    1. 'db' key in RegistryConfig,
+    2. RegistryConfig for explicitly defined keys
+    3. and DbAuth instance with which it was created.
```

Comment:
I don't think this is doing what it says.  In fact, I don't think it is or should be doing much more than `sqlalchemy.engine.URL`.  If you really want to add `checkDialectDriverExist()`, that's OK.

Reply:
Removed. `ConnectionStringFactory` is not a `sqlalchemy.engine.url.URL` factory effectively. Can the naming stay or would `URLBuilder` or `URLFactory`, or an even better suggestion, be more appropriate?